spec/controllers/events_controller_spec.rb

```diff
@@ -0,0 +1,89 @@
+require 'rails_helper'
+
+RSpec.describe EventsController, type: :controller do
+  let(:randsome) {
+    @randsome ||= double('Randsome',
+      random_attendee: { id: 'A41F1E1', name: name },
+      available_events: []
+    )
+  }
+
+  let(:base_meetup) {
+    @base_meetup ||= double('BaseMeetup',
+      self_events: [],
+      event: {},
+      attendance: [{ member: person }],
+      profile_image: img_name
+    )
+  }
+
+  let(:person) {
+    {
+      name: name,
+      photo: {
+        thumb: img_name
+      }
+    }
+  }
+
+  let(:name) { 'Mary' }
+
+  let(:img_name) { 'image.jpg' }
+
+  let(:event_payload) { { urlname: 'guru-sorocaba', event_id: '1b3b4fe' } }
+
+  before do
+    allow_any_instance_of(EventsController).to receive(:meetup).and_return(base_meetup)
+    allow_any_instance_of(EventsController).to receive(:randsome).and_return(randsome)
+    allow_any_instance_of(Randsome).to receive(:meetup).and_return(base_meetup)
+  end
+
+  describe 'GET #index' do
+    it 'assign events' do
+      get :index
+      expect(assigns(:events)).to match_array([])
+    end
+
+    it 'call the randsome service' do
+      expect(randsome).to receive(:available_events)
+      get :index
+    end
+  end
+
+  describe 'GET #show' do
+    it 'assign event' do
```

Comment:
No, thats actually because #show calls for `randsome.available_events` with return a empty as it is set in the line 7 and stubed in the line 37.